src/local.jl

```diff
@@ -18,15 +18,7 @@ function localzone()
     mask = Class(:STANDARD) | Class(:LEGACY)
 
     @static if Sys.isapple()
-        name = @mock read(`systemsetup -gettimezone`, String)  # Appears to only work as root
-        if startswith(name, "Time Zone: ")
-            name = strip(replace(name, "Time Zone: " => ""))
-        else
-            # link will be something like /usr/share/zoneinfo/Europe/Warsaw
-            name = @mock readlink("/etc/localtime")
-            name = match(r"(?<=zoneinfo/).*$", name).match
-        end
-        return TimeZone(name, mask)
+        return TimeZone(readlink("/etc/localtime")[27:end], mask)
```

Comment:
If you take the contents of the `else` the test should work again. I'll note I am not a fan of hard coding substrings like this as it's rather frail.

Reply:
I am experiencing the issue on Mac OS Catalina 10.15.4 with Visual Studio Code (works fine from the Terminal and VS Code has full disk access)
```
julia> using TimeZones
julia> localzone()
ERROR: IOError: could not spawn `systemsetup -gettimezone`: no such file or directory (ENOENT)
Stacktrace:
 [1] _spawn_primitive(::String, ::Cmd, ::Array{Any,1}) at ./process.jl:99
 [2] #550 at ./process.jl:112 [inlined]
 [3] setup_stdios(::Base.var"#550#551"{Cmd}, ::Array{Any,1}) at ./process.jl:196
 [4] _spawn at ./process.jl:111 [inlined]
 [5] open(::Cmd, ::Base.DevNull; write::Bool, read::Bool) at ./process.jl:370
 [6] open at ./process.jl:361 [inlined]
 [7] open(::Cmd, ::String, ::Base.DevNull) at ./process.jl:341
 [8] read(::Cmd) at ./process.jl:410
 [9] read(::Cmd, ::Type{String}) at ./process.jl:421
 [10] macro expansion at /Users/jbsc/.julia/packages/Mocking/U41JO/src/mock.jl:29 [inlined]
 [11] localzone() at /Users/jbsc/.julia/packages/TimeZones/OSWcm/src/local.jl:21
 ...
```
For example, for LibPQ.jl, I have to first precompile the package from the terminal and then I can call it within VS Code but could not precompile it from VS Code due to the error above (running VS Code as root didn't change that).

I also looked at
```
julia> read(`date +"%Z", String)
```

The symlink has the following syntax,
```
/var/db/timezone/zoneinfo/*
```
I could change it to 
```
match(r"(?<=/var/db/timezone/zoneinfo/).*", readlink("/etc/localtime")).match
```
but that would be equally frail.

Reply:
Interesting. The error `IOError: could not spawn `systemsetup -gettimezone`: no such file or directory (ENOENT)` implies that the process itself cannot be found. I'd be interested in the output from these commands under these conditions:
```julia
run(`which systemsetup`)
ENV["PATH"]
```
The regular expression in the code you removed can handle the example symlink you provided:
```julia
julia> name = "/var/db/timezone/zoneinfo/*"
"/var/db/timezone/zoneinfo/*"

julia> match(r"(?<=zoneinfo/).*$", name).match
"*"
```

Reply:
New environment. 
```
 [f269a46b] TimeZones v1.1.0
```
From VS Code (`julialang.language-julia` `v0.15.24`)

```
julia> run(`which systemsetup`)
ERROR: failed process: Process(`which systemsetup`, ProcessExited(1)) [1]

Stacktrace:
 [1] pipeline_error at ./process.jl:525 [inlined]
 [2] run(::Cmd; wait::Bool) at ./process.jl:440
 [3] run(::Cmd) at ./process.jl:438
 [4] top-level scope at REPL[9]:1
 [5] eval(::Module, ::Any) at ./boot.jl:331
 [6] eval_user_input(::Any, ::REPL.REPLBackend) at /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.4/REPL/src/REPL.jl:86
 [7] run_backend(::REPL.REPLBackend) at /Users/jbsc/.julia/packages/Revise/C272c/src/Revise.jl:1075
 [8] top-level scope at none:0
```

```
julia> ENV["PATH"]
ERROR: KeyError: key "PATH" not found
Stacktrace:
 [1] (::Base.var"#459#460")(::String) at ./env.jl:79
 [2] access_env at ./env.jl:43 [inlined]
 [3] getindex(::Base.EnvDict, ::String) at ./env.jl:79
 [4] top-level scope at REPL[10]:1
 [5] eval(::Module, ::Any) at ./boot.jl:331
 [6] eval_user_input(::Any, ::REPL.REPLBackend) at /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.4/REPL/src/REPL.jl:86
 [7] run_backend(::REPL.REPLBackend) at /Users/jbsc/.julia/packages/Revise/C272c/src/Revise.jl:1075
 [8] top-level scope at none:0
```

Reply:
Thank you. Ok the VS Code environment is very restrictive. I'm completely fine with relying on using `/etc/localtime` for time zone detection as `systemsetup` only actually works if you are running as `root` which will be very rare:
```
$ systemsetup -gettimezone; echo $?
You need administrator access to run this tool... exiting!
0
```

Reply:
This change should work and allow the tests to work:
```suggestion
        # Link will point to something like "/usr/share/zoneinfo/Europe/Warsaw"
        link = @mock readlink("/etc/localtime")
        name = match(r"(?<=zoneinfo/).*$", link).match
        return TimeZone(name, mask)
```